### PR TITLE
Now disables all GraphicBuffer optimizations when building without flag

### PR DESCRIPTION
**Pull request by emilwestergren**
@ziquanyu peer review

### CODE REVIEW

File: source/draw/gpu/opengl/AndroidContext.ooc [hunk 1]
```diff
@@ -60,67 +60,82 @@ AndroidContext: class extends OpenGLContext {
...
+		result: RasterImage
+		version(optiGraphicbufferupload) {
+			result = match (source) {
+				case (image : OpenGLMonochrome) =>
+					this isAligned(image channels * image size x) ? this toRaster(image) : super(image)
+				case (image : OpenGLUv) =>
+					this isAligned(image channels * image size x) ? this toRaster(image) : super(image)
+				case => super(source)
+			}
+		} else {
```
Comment (marcusnaslund) [inline source/draw/gpu/opengl/AndroidContext.ooc]: No brackets needed

File: source/draw/gpu/opengl/AndroidContext.ooc [hunk 2]
```diff
@@ -60,67 +60,82 @@ AndroidContext: class extends OpenGLContext {
...
-		match (source) {
-			case (image : OpenGLMonochrome) =>
-				this isAligned(image channels * image size x) ? this toRaster(image) : super(image)
-			case (image : OpenGLUv) =>
-				this isAligned(image channels * image size x) ? this toRaster(image) : super(image)
-			case => super(source)
+		result: RasterImage
+		version(optiGraphicbufferupload) {
+			result = match (source) {
+				case (image : OpenGLMonochrome) =>
```
Comment (marcusnaslund) [inline source/draw/gpu/opengl/AndroidContext.ooc]: Not your fault *this* time, but it got merged 7 months ago, so I'm off the hook as well. No colon before `:`.

File: source/draw/gpu/opengl/AndroidContext.ooc [hunk 3]
```diff
@@ -60,67 +60,82 @@ AndroidContext: class extends OpenGLContext {
...
+				padding := targetImage uvPadding % targetImage stride
+				this packToRgba(sourceImage y, targetImageRgba, IntBox2D new(0, 0, targetWidth, targetImage y size y), padding)
+				this packToRgba(sourceImage uv, targetImageRgba, IntBox2D new(0, targetImageRgba size y - targetImage uv size y, targetWidth, targetImage uv size y), padding)
+				result = OpenGLPromise new(this)
+				(result as OpenGLPromise) sync()
+				targetImageRgba referenceCount decrease()
+			} else {
+				result = super(source, target)
+			}
+		} else {
```
Comment (marcusnaslund) [inline source/draw/gpu/opengl/AndroidContext.ooc]: No brackets needed.

File: source/draw/gpu/opengl/AndroidContext.ooc [hunk 4]
```diff
@@ -60,67 +60,82 @@ AndroidContext: class extends OpenGLContext {
...
-		else
+		version(optiGraphicbufferupload) {
+			aligned := this isAligned(gpuImage size x)
+			if (aligned && gpuImage instanceOf(OpenGLMonochrome))
+				result = this _toRasterAsync(gpuImage as OpenGLMonochrome)
+			else if (aligned && gpuImage instanceOf(OpenGLUv))
+				result = this _toRasterAsync(gpuImage as OpenGLUv)
+			else
+				result = super(gpuImage)
+		} else {
```
Comment (marcusnaslund) [inline source/draw/gpu/opengl/AndroidContext.ooc]: No brackets needed.

File: source/draw/gpu/opengl/AndroidContext.ooc [hunk 5]
```diff
@@ -60,67 +60,82 @@ AndroidContext: class extends OpenGLContext {
...
-			result = this toRasterAsync(gpuImage as OpenGLUv)
-		else
+		version(optiGraphicbufferupload) {
+			aligned := this isAligned(gpuImage size x)
+			if (aligned && gpuImage instanceOf(OpenGLMonochrome))
+				result = this _toRasterAsync(gpuImage as OpenGLMonochrome)
+			else if (aligned && gpuImage instanceOf(OpenGLUv))
+				result = this _toRasterAsync(gpuImage as OpenGLUv)
+			else
+				result = super(gpuImage)
```
Comment (marcusnaslund) [inline source/draw/gpu/opengl/AndroidContext.ooc]: Seems like there should be a `match` statement here instead.

**Comment by marcusnaslund** [comment]
Done.

File: source/draw/gpu/opengl/AndroidContext.ooc (hunk 2 above)
Comment (emilwestergren) [inline source/draw/gpu/opengl/AndroidContext.ooc]: You mean no space?

Comment (marcusnaslund) [inline source/draw/gpu/opengl/AndroidContext.ooc]: Heh, yes.

**Comment by emilwestergren** [comment]
Updated

**Comment by marcusnaslund** [comment]
Looks good.

**Comment by marcusnaslund** [comment]
Noticeable difference on my non-snapdragon build from PRInt! :) I should probably add this flag to its settings.

**Pull request merged**